packages/web3-eth/src/eth_tx.ts

```diff
@@ -0,0 +1,315 @@
+	CommonOrChainAndHardforkError,
+	Eip1559GasPriceError,
+	Eip1559NotSupportedError,
+	InvalidGasOrGasPrice,
+	InvalidMaxPriorityFeePerGasOrMaxFeePerGas,
+	InvalidNonceOrChainIdError,
+	InvalidTransactionObjectError,
+	MissingChainOrHardforkError,
+	MissingCustomChainError,
+	MissingCustomChainIdError,
+	MissingGasError,
+	UnableToPopulateNonceError,
+	UnsupportedFeeMarketError,
+	UnsupportedTransactionTypeError,
+} from './errors';
+import { PopulatedUnsignedTransaction, Transaction } from './types';
+
+export function formatTransaction<
+	DesiredType extends ValidTypes = ValidTypes,
+	ReturnType = ValidReturnTypes[DesiredType],
+>(
+	transaction: Transaction,
+	desiredType: DesiredType,
+	overrideMethod?: (transaction: Transaction) => Transaction<ReturnType>,
+): Transaction<ReturnType> {
+	if (overrideMethod !== undefined) return overrideMethod(transaction);
+	const formattedTransaction = {
+		...transaction,
+		value:
+			transaction.value === '0x' ? '0x' : convertToValidType(transaction.value, desiredType),
+		gas: convertToValidType(transaction.gas, desiredType),
+		gasPrice: convertToValidType(transaction.gasPrice, desiredType),
+		type: convertToValidType(transaction.type, desiredType),
+		maxFeePerGas: convertToValidType(transaction.maxFeePerGas, desiredType),
+		maxPriorityFeePerGas: convertToValidType(transaction.maxPriorityFeePerGas, desiredType),
+		nonce: convertToValidType(transaction.nonce, desiredType),
+		chainId: convertToValidType(transaction.chainId, desiredType),
+		gasLimit: convertToValidType(transaction.gasLimit, desiredType),
+		v: convertToValidType(transaction.v, desiredType),
+		common: {
+			...transaction.common,
+			customChain: {
+				...transaction.common?.customChain,
+				networkId: convertToValidType(
+					transaction.common?.customChain?.networkId,
+					desiredType,
+				),
+				chainId: convertToValidType(transaction.common?.customChain?.chainId, desiredType),
+			},
+		},
+	};
+	// TODO - TSC is complaining that ReturnType could be instantiated with an
+	// arbitrary type which could be unrelated to 'string | number | bigint | undefined'
+	return formattedTransaction as unknown as Transaction<ReturnType>;
+}
+
+export const detectTransactionType = (
+	transaction: Transaction,
+	overrideMethod?: (transaction: Transaction) => Numbers | undefined,
+): Numbers | undefined => {
+	if (overrideMethod !== undefined) return overrideMethod(transaction);
+	if (transaction.type !== undefined) return transaction.type;
+
+	if (transaction.maxFeePerGas !== undefined || transaction.maxPriorityFeePerGas !== undefined)
+		return '0x2';
+	if (transaction.hardfork === 'london') return '0x2';
+	if (transaction.common?.hardfork === 'london') return '0x2';
+
+	if (transaction.accessList !== undefined) return '0x1';
+	if (transaction.hardfork === 'berlin') return '0x1';
+	if (transaction.common?.hardfork === 'berlin') return '0x1';
+
+	return undefined;
+};
+
+const validateCustomChainInfo = (transaction: Transaction) => {
```

Comment:
I think it should return boolean, instead of user relying on error throw for detection of validation

Reply:
The default behaviour of validation in the whole library is to throw error, so that's seems fine. 

In case decide to return boolean value then would better to rename it to `isValidCustomChainInfo`

Reply:
This is an internal method that's used only for `validateTransactionForSigning`, so I think it would make since to error out. We could add `isValidCustomChainInfo` in the future if it's needed, but as @nazarhussain said, it's the library default to error for `validate_` methods

Throwing errors and returning `void` allows for the developer to just call `validate_(input)` and carry on instead of having to always add a check e.g. `if(!validate_(input))`

Reply:
Its recommended to never use try/catch for **control logic**. 
Currently in our case: if not valid throw error else donot throw error
Reason: performance, good pattern,